src/app/login/login.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { AuthService } from '../core/auth/auth.service';
+import { IUserCredentials } from '../core/auth/auth.interfaces';
+
+@Component({
+    selector: 'sm-login',
+    styles: [require('./login.component.scss')],
+    template: require('./login.component.html')
+})
+export class LoginComponent {
+    private credentials: IUserCredentials = {
+        email: null,
+        password: null
+    };
+    private loggingError: boolean = false;
+
+    constructor(private auth: AuthService, private router: Router) {}
+
+    login() {
+        this.auth.login(this.credentials)
+            .subscribe(() => {
```

Comment:
If you subscribe to event stream, you should unsubscribe when component destroyed